nvbench/linear_axis_space.cxx

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2022 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 with the LLVM exception
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.
+ *
+ *  You may obtain a copy of the License at
+ *
+ *      http://llvm.org/foundation/relicensing/LICENSE.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "linear_axis_space.cuh"
+
+#include <nvbench/type_axis.cuh>
+
+namespace nvbench
+{
+
+linear_axis_space::linear_axis_space(std::size_t in_index,
+                                     std::size_t out_index)
+    : iteration_space_base({in_index}, {out_index})
+{}
+
+linear_axis_space::~linear_axis_space() = default;
+
+detail::axis_space_iterator linear_axis_space::do_get_iterator(axes_info info) const
+{
+  std::size_t loc(m_output_indices[0]);
```

Comment:
This is just asking to get [vexxed](https://en.wikipedia.org/wiki/Most_vexing_parse) ;) 

```suggestion
  std::size_t loc{m_output_indices[0]};
```